torch/nested/_internal/sdpa.py

```diff
@@ -736,7 +737,7 @@ def jagged_scaled_dot_product_attention(
         return ViewNestedFromBuffer.apply(
             attention.squeeze(0), output_nt_info["offsets"]
         ).transpose(1, 2)
-    elif backend_choice == SDPBackend.MATH:
+    elif backend_choice == 3:
```

Comment:
Changes:
- Instead of using the SDPBackend enum, which isn't supported by dynamo, use ints 
- Construct the NestedTensor in an allow_in_graph region, otherwise the construction is traced away, and we return a plain Tensor
- _select_sdp_backend is wrapped in allow_in_graph because SDPAParams construction results in graph break
- global state accesses are done outside of the allow_in_graph region (not sure if this matters)